### PR TITLE
expression: implement vectorized evaluation for `builtinTruncateRealSig`

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
expression: implement vectorized evaluation for `builtinTruncateRealSig`,for #12105 

### What is changed and how it works?
```
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinMathFunc/builtinTruncateRealSig-VecBuiltinFunc-8         	  200000	      8079 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinTruncateRealSig-NonVecBuiltinFunc-8      	   50000	     36437 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by SunRunAway** [comment]
@tsthght Please resolve the conflict, Thanks.

File: expression/builtin_math_vec.go [hunk 1]
```diff
@@ -465,6 +491,10 @@ func (b *builtinAbsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 	return nil
 }
 
+func (b *builtinTruncateRealSig) vectorized() bool {
```
Comment (Reminiscent) [inline expression/builtin_math_vec.go]: It's better to put the ```func (b *builtinTruncateRealSig) vectorized() bool``` and ```func (b *builtinTruncateRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error``` together.

Comment (tsthght) [reply expression/builtin_math_vec.go]: ok

**Pull request merged**